src/appleseed.studio/python/pythoninterpreter.h

```diff
@@ -0,0 +1,58 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Gleb Mishchenko, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_STUDIO_PYTHON_INTERPRETER_H
+#define APPLESEED_STUDIO_PYTHON_INTERPRETER_H
+
+// appleseed.foundation headers.
+#include <foundation/core/concepts/noncopyable.h>
+
+namespace appleseed {
+namespace studio {
+
+// Forward declarations.
+class OutputRedirector;
+
+class PythonInterpreter
+  : public foundation::NonCopyable
+{
+  public:
+    static PythonInterpreter& instance();
+    void execute_command(const char* command);
+
+    void redirect_output(OutputRedirector redirector);
```

Comment:
I thought about this and I'm not sure. It's called like this:
PythonInterpreter::instance().redirect_output(OutputRedirector(output));

So I guess it will be destroyed right after the call? If yes, the pass-by-value is correct here